src/metpy/remote/aws.py

```diff
@@ -0,0 +1,711 @@
+# Copyright (c) 2023 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Tools for reading known collections of data that are hosted on Amazon Web Services (AWS)."""
+import bisect
+from datetime import datetime, timedelta, timezone
+import itertools
+from pathlib import Path
+import shutil
+
+import xarray as xr
+
+from ..io import Level2File, Level3File
+from ..package_tools import Exporter
+
+exporter = Exporter(globals())
+
+
+def ensure_timezone(dt):
+    """Add UTC timezone if no timezone present."""
+    return dt.replace(tzinfo=timezone.utc) if dt.tzinfo is None else dt
```

Comment:
Should we alert users if we modify timezone assumptions for provided `dt`? Or at least document in `get_product`

Reply:
I mean, it's an error if we don't do this, but sure, we can say that `dt` should have the correct timezone or UTC will be assumed.

Reply:
For the majority of users that may end up being more of an annoyance than useful, unless we're pushing the "always specify time zone" agenda particularly hard. Documenting might be enough for most folks.